cfme/tests/infrastructure/test_timelines.py

```diff
@@ -103,99 +152,215 @@ def _power_on(self):
-    for event in events_list:
-        vm_event = VMEvent(new_vm, event)
-        logger.info('Will generate event %r on machine %r', event, new_vm.name)
-        wait_for(vm_event.emit, timeout='5m', message='Event {} did timeout'.format(event))
-        vm_event.catch_in_timelines(soft_assert, targets)
+    logger.info('Will generate event %r on machine %r', event, new_vm.name)
+    wait_for(vm_event.emit, timeout='7m', message='Event {evt}failed'.format(evt=event))
+    vm_event.catch_in_timelines(soft_assert, targets)
 
 
-def test_infra_diagnostic_timeline_events(new_vm, soft_assert, mark_vm_as_appliance):
-    events_list = ['create']
+@pytest.mark.meta(blockers=[BZ(1542962, forced_streams=['5.8'])])
+def test_infra_timeline_policy_event(new_vm, control_policy, soft_assert):
+    event = 'policy'
+    targets = (new_vm, new_vm.host, new_vm.cluster, new_vm.provider)
+    vm_event = VMEvent(new_vm, event)
+    logger.info('Will generate event %r on machine %r', event, new_vm.name)
+    wait_for(vm_event.emit, timeout='7m', message='Event {} did timeout'.format(event))
+    vm_event.catch_in_timelines(soft_assert, targets, policy_events=True)
+
+
+def test_infra_timeline_stop_event(new_vm, soft_assert):
+    event = 'stop'
+    targets = (new_vm, new_vm.cluster, new_vm.host, new_vm.provider)
+    vm_event = VMEvent(new_vm, event)
+    logger.info('Will generate event %r on machine %r', event, new_vm.name)
+    wait_for(vm_event.emit, timeout='7m', message='Event {} failed'.format(event))
+    vm_event.catch_in_timelines(soft_assert, targets)
+
+
+def test_infra_timeline_start_event(new_vm, soft_assert):
+    event = 'start'
+    targets = (new_vm, new_vm.host)
+    vm_event = VMEvent(new_vm, event)
+    logger.info('Will generate event %r on machine %r', event, new_vm.name)
+    wait_for(vm_event.emit, timeout='7m', message='Event {} failed'.format(event))
+    vm_event.catch_in_timelines(soft_assert, targets)
+
+
+def test_infra_timeline_suspend_event(new_vm, soft_assert):
+    event = 'suspend'
+    targets = (new_vm, new_vm.cluster, new_vm.host, new_vm.provider)
+    vm_event = VMEvent(new_vm, event)
+    logger.info('Will generate event %r on machine %r', event, new_vm.name)
+    wait_for(vm_event.emit, timeout='7m', message='Event {} failed'.format(event))
+    vm_event.catch_in_timelines(soft_assert, targets)
+
+
+def test_infra_timeline_diagnostic(new_vm, soft_assert, mark_vm_as_appliance):
+    event = 'create'
     targets = (new_vm.appliance.server,)
-    for event in events_list:
-        logger.info('Will generate event %r on machine %r', event, new_vm.name)
-        vm_event = VMEvent(new_vm, event)
-        vm_event.catch_in_timelines(soft_assert, targets)
+    logger.info('Will generate event %r on machine %r', event, new_vm.name)
+    vm_event = VMEvent(new_vm, event)
+    vm_event.catch_in_timelines(soft_assert, targets)
+
+
+@pytest.mark.provider([VMwareProvider], override=True, scope='function')
```

Comment:
please add short comments explaining why some test is started only against some certain provider type

Reply:
I add explaination for one, and for the other one i removed it it was not needed anymore.

Reply:
I don't see where you added comment